readers/api/requests.go

```diff
@@ -39,5 +41,53 @@ func (req listMessagesReq) validate() error {
 		return apiutil.ErrInvalidComparator
 	}
 
+	validAggregations := map[string]bool{
+		"MAX":   true,
+		"MIN":   true,
+		"AVG":   true,
+		"SUM":   true,
+		"COUNT": true,
+		"max":   true,
+		"min":   true,
+		"avg":   true,
+		"sum":   true,
+		"count": true,
+	}
+
+	aggregationValid := validAggregations[req.pageMeta.Aggregation]
+
```

Comment:
Let's keep this simple to something like this

```go
	validAggregations := []string{"MAX", "MIN", "AVG", "SUM", "COUNT"}

	if req.pageMeta.Aggregation != "" {
		if !slices.Contains(validAggregations, strings.ToUpper(req.pageMeta.Aggregation)) {
			return apiutil.ErrInvalidAggregation
		}
	}
```
You can change  `validAggregations` to a global internal constant in the API package